### PR TITLE
feat: compatibility with Express.js, Koa, Fastify & NestJS

### PR DESCRIPTION
**Pull request by romain-gilliotte**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)
- [ ] Ensure that Types have been updated according to your changes (if needed)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/_example/src/frameworks/nest-express-v8.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { Module } from '@nestjs/common';
+import { NestFactory } from '@nestjs/core';
+
+@Module({ imports: [], controllers: [], providers: [] })
```
Comment (arnaud-moncel) [inline packages/_example/src/frameworks/nest-express-v8.ts:4]: I think we can add simple route like the other frameworks? What do you think?
```suggestion
@Controller('/')
class BasicController {
  @Get()
  findAll() {
    return { error: null, framework: 'Nest fastify' };
  }
}

@Module({ imports: [], controllers: [BasicController], providers: [] })
```

File: packages/_example/src/frameworks/nest-fastify-v8.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { FastifyAdapter, NestFastifyApplication } from '@nestjs/platform-fastify';
+import { Module } from '@nestjs/common';
+import { NestFactory } from '@nestjs/core';
+
+@Module({ imports: [], controllers: [], providers: [] })
```
Comment (arnaud-moncel) [inline packages/_example/src/frameworks/nest-fastify-v8.ts:5]: Same here 😄

File: patches/@fastify+express+1.1.0.patch [hunk 3]
```diff
@@ -0,0 +1,26 @@
+diff --git a/node_modules/@fastify/express/index.d.ts b/node_modules/@fastify/express/index.d.ts
+deleted file mode 100644
+index 42fb81a..0000000
+--- a/node_modules/@fastify/express/index.d.ts
```
Comment (arnaud-moncel) [inline patches/@fastify+express+1.1.0.patch:4]: It seem this is not necessary no ?

File: packages/agent/test/builder/agent_frameworks.test.ts [hunk 4]
```diff
@@ -0,0 +1,147 @@
+/* eslint-disable max-classes-per-file */
+import { FastifyAdapter, NestFastifyApplication } from '@nestjs/platform-fastify';
```
Comment (arnaud-moncel) [inline packages/agent/test/builder/agent_frameworks.test.ts:2]: nice tests 😍 , i'm wondering if it is possible to test if the cors doesn't affect user app route

**Review by arnaud-moncel — COMMENTED** [review]
Nice Job 💪, I have some comments feel free to discuss them

File: packages/_example/src/frameworks/nest-express-v8.ts (hunk 1 above)
Comment (romain-gilliotte) [reply packages/_example/src/frameworks/nest-express-v8.ts:4]: Good idea
To be honest, I did not know how to make a route.

File: patches/@fastify+express+1.1.0.patch (hunk 3 above)
Comment (romain-gilliotte) [reply patches/@fastify+express+1.1.0.patch:4]: The project does not compile on my machine if I don't delete this typing file (it wants me to install the Fastify dependency on `@forestadmin/agent`)

Does it works on your machine?

File: packages/_example/package.json [hunk 5]
```diff
@@ -12,10 +12,21 @@
     "@forestadmin/datasource-sql": "1.0.0-beta.26",
     "@forestadmin/datasource-toolkit": "1.0.0-beta.16",
     "@forestadmin/datasource-mongoose": "1.0.0-beta.1",
+    "@koa/router": "^10.1.1",
+    "@nestjs/common": "^8.4.5",
+    "@nestjs/core": "^8.4.5",
+    "@nestjs/platform-express": "^8.4.5",
```
[the remark this answers is not in the document]
Comment (romain-gilliotte) [reply packages/_example/package.json:18]: it is a peer depency of @nestjs/core

File: packages/agent/test/builder/agent_frameworks.test.ts (hunk 4 above)
Comment (romain-gilliotte) [reply packages/agent/test/builder/agent_frameworks.test.ts:2]: Currently, our `cors` are wide open...
https://github.com/ForestAdmin/agent-nodejs/blob/beta/packages/agent/src/agent/forestadmin-http-driver.ts#L29

At some point I wanted to restrict then to *.forestadmin.com but I was afraid to break the white labeling feature.
Do you know how it works in agent v1?

**Review by arnaud-moncel — APPROVED** [review]
🚀

**Pull request merged**